controllers/ingressnodefirewallconfig_controller_test.go

```diff
@@ -41,10 +45,16 @@ var _ = Describe("Ingress nodefirewall config Controller", func() {
 				return err
 			}, 2*time.Second, 200*time.Millisecond).ShouldNot((HaveOccurred()))
 			Expect(daemonSet).NotTo(BeZero())
-			Expect(len(daemonSet.Spec.Template.Spec.Containers)).To(BeNumerically(">", 0))
-			for _, c := range daemonSet.Spec.Template.Spec.Containers {
-				image, ok := daemonContainers[c.Name]
-				Expect(ok).To(BeTrue(), fmt.Sprintf("container %s not found in %s", c.Name, daemonContainers))
+			Expect(daemonSet.Spec.Template.Spec.InitContainers).To(HaveLen(len(daemonInitContainers)))
+			Expect(daemonSet.Spec.Template.Spec.Containers).To(HaveLen(len(daemonContainers)))
+			for name, image := range daemonInitContainers {
+				ok, c := findContainerByName(name, daemonSet.Spec.Template.Spec.InitContainers)
```

Comment:
you don't really need another loop here daemonCOntainers is hash map u provide the name and u get the result ?

Reply:
Not sure I understand; I want iterate over the expected containers and verify that each named container exists in `Spec.InitContainers`, which is a slice.

Reply:
> you don't really need another loop here daemonCOntainers is hash map u provide the name and u get the result ?

Putting it another way: we should iterate over the expectations. The way the test was previously allowed me to add the init container reference "mount-bpffs" to `daemonContainers`, yet the test didn't fail. And I got caught out by that because it should have failed.

Reply:
`findContainerByName` is doing loop for no reason u can lookup the map by name and get ur answer ?

Reply:
OK - as we have the new len assertion we can iterate as it was previously.

Reply:
Done in https://github.com/openshift/ingress-node-firewall/compare/46f5a27d3c523a4ad2439511ddeaaefcb6dfcb92..ec4fb75a0efe555f259c49584a1a10633ce82b91.